src/main/java/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -46,14 +56,32 @@ public Study getStudy(String appId, String studyId, boolean throwsException) {
      * so we can provide a cache for these infrequently changing identifiers.
      */
     public Set<String> getStudyIds(String appId) {
-        return getStudies(appId, false).stream()
-                .map(Study::getId).collect(BridgeCollectors.toImmutableSet());
+        int offset = 0;
+        Set<String> identifiers = new HashSet<>();
+        PagedResourceList<Study> results = null;
+        
+        do {
+            results = getStudies(appId, offset, API_MAXIMUM_PAGE_SIZE, false);
+            identifiers.addAll(results.getItems().stream().map(Study::getIdentifier).collect(toSet()));
+            offset += API_MAXIMUM_PAGE_SIZE;
+        } while(identifiers.size() < results.getTotal());
```

Comment:
There's not really any reason to use pagination here. We're requesting the same amount of data from the database no matter what, except with the extra overhead of making a bunch of paginated calls, and the extra complexity of adding server logic for something the database already handles natively.

Reply:
Removed (no paging indicated to SQL)